syntax = "proto3";

package test;

message MixedInput {
    enum Task {
        TXT2IMG = 0;
        IMG2IMG = 1;
        INPAINT = 2;
    }

    bytes image_data = 1;
    string prompt = 2;
    optional int32 height = 3;
    optional int32 width = 4;
    float scale = 5;
    Task task = 6;
    repeated string lora_names = 7;
}

message MixedOutput {
    string model_name = 1;
    bytes generated_image = 2;
}